editor/store/effects.js

```diff
@@ -323,34 +323,36 @@ export default {
 		const { dispatch } = store;
 
 		let result;
-		if ( id ) {
+		if ( id && wp.api.models.Blocks ) {
 			result = new wp.api.models.Blocks( { id } ).fetch();
-		} else {
+		} else if ( wp.api.collections.Blocks ) {
 			result = new wp.api.collections.Blocks().fetch();
 		}
 
-		result.then(
-			( reusableBlockOrBlocks ) => {
-				dispatch( {
-					type: 'FETCH_REUSABLE_BLOCKS_SUCCESS',
-					id,
-					reusableBlocks: castArray( reusableBlockOrBlocks ).map( ( { id: itemId, title, content } ) => {
-						const [ { name: type, attributes } ] = parse( content );
-						return { id: itemId, title, type, attributes };
-					} ),
-				} );
-			},
-			( error ) => {
-				dispatch( {
-					type: 'FETCH_REUSABLE_BLOCKS_FAILURE',
-					id,
-					error: error.responseJSON || {
-						code: 'unknown_error',
-						message: __( 'An unknown error occurred.' ),
-					},
-				} );
-			}
-		);
```

Comment:
I think these three methods would read nicer if we perform these checks at the start rather than indenting everything in an `if`. For example:

```js
FETCH_REUSABLE_BLOCKS( action, store ) {
    const { id } = action;
    const { dispatch } = store;

    if ( ! wp.api.models.Blocks || ! wp.api.collections.Blocks) {
        return;
    }

    ...
```

Reply:
👍